components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/ServiceURLBuilder.java

```diff
@@ -112,6 +112,18 @@ default ServiceURLBuilder setOrganization(String orgId) {
      *
      * @return {@link ServiceURL}.
      * @throws URLBuilderException If error occurred while constructing the URL.
+     * @deprecated Use {@link #buildURL(String)} instead.
      */
+    @Deprecated
     ServiceURL build() throws URLBuilderException;
+
+    /**
+     * Returns a ServiceURL with the protocol, hostname, port, proxy context path, a web context
+     * root and the tenant domain (appended if required).
+     *
+     * @param hostname Hostname.
+     * @return {@link ServiceURL}.
+     * @throws URLBuilderException If error occurred while constructing the URL.
+     */
+    ServiceURL build(String hostname) throws URLBuilderException;
```

Comment:
Can we make this method as default cause some extension may fail because of method param mismatch

Reply:
fixed with 0f13202